operators/src/mock/mock_raster_source.rs

```diff
@@ -109,18 +109,90 @@ where
         _ctx: &'a dyn crate::engine::QueryContext,
     ) -> Result<futures::stream::BoxStream<crate::util::Result<RasterTile2D<Self::RasterType>>>>
     {
-        let inner_stream = stream::iter(
-            self.data
-                .iter()
-                .filter(move |t| {
-                    t.time.intersects(&query.time_interval)
-                        && t.tile_information()
-                            .spatial_partition()
-                            .intersects(&query.spatial_bounds)
-                })
-                .cloned()
-                .map(Result::Ok),
-        );
+        let mut parts: Vec<RasterTile2D<T>> = self
+            .data
+            .iter()
+            .filter(move |t| {
+                t.time.intersects(&query.time_interval)
+                    && t.tile_information()
+                        .spatial_partition()
+                        .intersects(&query.spatial_bounds)
+            })
+            .cloned()
+            .collect();
+
+        // This is a workaround to avoid errors when a provider does not fill the complete query rectangle.
```

Comment:
what does the mock raster source have to do with providers?

Reply:
but also here: wouldn't it make more sense to insert the fillers at creation time (of the source) instead of query time?

Reply:
i removed the fillers and now all providers can provide known times covering the query time. If they don't the gdal source will fallback to the query time and log warnings (no error since it is not wrong only unfortunate wrt the cache)